src/onyx/api.clj

```diff
@@ -145,40 +145,52 @@
         args (add-percentages-to-log-entry config job args tasks (:catalog job) id)]
     (create-log-entry :submit-job args)))
 
+(defn validate-submission [job peer-client-config]
+  (try
+    (validator/validate-peer-client-config peer-client-config)
+    (validator/validate-job job)
+    (validator/validate-flow-conditions job)
```

Comment:
I believe we can drop all these additional cases if validate-job did its job correctly. I've fixed this in a commit I'll push up.